example/snippets.cpp

```diff
@@ -1530,6 +1601,637 @@ void section_connection_pool(string_view server_hostname, string_view username,
+            //<-
+        );
+        //->
+        //<-
+        ASSERT(
+            // clang-format off
+            //->
+            format_sql(opts, "SELECT {} FROM t", identifier("mytable", "myfield")) == "SELECT `mytable`.`myfield` FROM t"
+            //<-
+            // clang-format on
+        );
+        //->
+        //<-
+        ASSERT(
+            // clang-format off
+            //->
+            format_sql(opts, "SELECT {} FROM t", identifier("mydb", "mytable", "myfield")) == "SELECT `mydb`.`mytable`.`myfield` FROM t"
+            //<-
+            // clang-format on
+        );
+        //->
+        //]
+    }
+
+    // Advanced section
+    {
+        //[sql_formatting_formatter_use
+        // We can now use employee as a built-in value
+        std::string query = boost::mysql::format_sql(
+            conn.format_opts().value(),
+            "INSERT INTO employee (first_name, last_name, company_id) VALUES ({}), ({})",
+            employee_t{"John", "Doe", "HGS"},
+            employee_t{"Rick", "Johnson", "AWC"}
+        );
+
+        ASSERT(
+            query ==
+            "INSERT INTO employee (first_name, last_name, company_id) VALUES "
+            "('John', 'Doe', 'HGS'), ('Rick', 'Johnson', 'AWC')"
+        );
+        //]
+
+        conn.execute(query, r);
+    }
+    {
+        const auto opts = conn.format_opts().value();
+
+        //[sql_formatting_auto_indexing
+        ASSERT(format_sql(opts, "SELECT {}, {}, {}", 42, "abc", nullptr) == "SELECT 42, 'abc', NULL");
+        //]
+    }
+    {
+        const auto opts = conn.format_opts().value();
+
+        //[sql_formatting_manual_auto_mix
+        try
+        {
+            // Mixing manual and auto indexing is illegal. This will throw an exception.
+            format_sql(opts, "SELECT {0}, {}", 42);
+            //<-
+            ASSERT(false);
+            //->
+        }
+        catch (const boost::system::system_error& err)
+        {
+            ASSERT(err.code() == boost::mysql::client_errc::format_string_manual_auto_mix);
+        }
+        //]
+    }
+    {
+        const auto opts = conn.format_opts().value();
+
+        //[sql_formatting_unused_args
+        // This is OK
+        std::string query = format_sql(opts, "SELECT {}", 42, "abc");
+        //]
+        ASSERT(query == "SELECT 42");
+    }
+    {
+        const auto opts = conn.format_opts().value();
+
+        //[sql_formatting_brace_literal
+        ASSERT(format_sql(opts, "SELECT 'Brace literals: {{ and }}'") == "SELECT 'Brace literals: { and }'");
+        //]
+    }
+    {
+        const auto opts = conn.format_opts().value();
+
+        //[sql_formatting_format_double_error
+        try
+        {
+            // We're trying to format a double infinity value, which is not
+            // supported by MySQL. This will throw an exception.
+            format_sql(opts, "SELECT {}", HUGE_VAL);
```

Comment:
this snippet, although meant to show exceptions, but it could at least try to assign the returned string to indicate that format_sql is meant to return a string.

in-particular it shows the difference compared to `format_sql_to`